pyanaconda/ui/gui/spokes/root_password.py

```diff
@@ -157,56 +183,97 @@ def sensitive(self):
+        if not error_message:
+            # all is fine, just clear the message
+            self.clear_info()
+        elif not self.password and not self.password_confirmation:
+            # Clear any info message if both the password and password
+            # confirmation fields are empty.
+            # This shortcut is done to make it possible for the user to leave the spoke
+            # without inputting any root password. Separate logic makes sure an
+            # empty string is not set as the root password.
+            self.clear_info()
+        else:
+            if self.checker.policy.strict or unwaivable_check_failed:
+                # just forward the error message
+                self.show_warning_message(error_message)
+            else:
+                # add suffix for the click twice logic
+                self.show_warning_message("%s %s" % (error_message, _(constants.PASSWORD_DONE_TWICE)))
+
+        # Next check if the user can escape from the root spoke or stay forever !
+
+        # Depending on the policy we allow users to waive the password strength
+        # and non-ASCII checks. If the policy is set to strict, the password
+        # needs to be strong, but can still contain non-ASCII characters.
+        self.can_go_back = False
+        self.needs_waiver = True
+
+        # This shortcut is done to make it possible for the user to leave the spoke
+        # without inputting any root password. Separate logic makes sure an
+        # empty string is not set as the root password.
+        if not self.password and not self.password_confirmation:
+            self.can_go_back = True
+            self.needs_waiver = False
+        elif self.checker.success:
+            # if all checks were successful we can always go back to the hub
+            self.can_go_back = True
+            self.needs_waiver = False
+        elif unwaivable_check_failed:
+            self.can_go_back = False
+        else:
+            if self.checker.policy.strict:
+                if not self._validity_check.result.success:
+                    # failing validity check in strict
+                    # mode prevents us from going back
+                    self.can_go_back = False
+                elif not self._ascii_check.result.success:
+                    # but the ASCII check can still be waived
+                    self.can_go_back = True
+                    self.needs_waiver = True
+                else:
+                    self.can_go_back = True
+                    self.needs_waiver = False
+            else:
+                if not self._validity_check.result.success:
+                    self.can_go_back = True
+                    self.needs_waiver = True
+                elif not self._ascii_check.result.success:
+                    self.can_go_back = True
+                    self.needs_waiver = True
+                else:
+                    self.can_go_back = True
+                    self.needs_waiver = False
```

Comment:
I don't like this big function and also `if-else-if-elif....`.

Could you please split these to multiple functions?